rewrite-java/src/main/java/org/openrewrite/java/cleanup/ChainStringBuilderAppendCalls.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.java.utils.ExpressionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+public class ChainStringBuilderAppendCalls extends Recipe {
+    private static final MethodMatcher STRING_BUILDER_APPEND = new MethodMatcher("java.lang.StringBuilder append(String)");
```

Comment:
Can't we also match all other `append()` overloads with a single parameter?

Reply:
good point, but for other `append()`  methods with other kinds of parameters, we don't want to transform.
Like this
```java
    void method1() {
        StringBuilder sb = new StringBuilder();
        int x = 1;
        sb.append(1 + x + 2);
    }
```
what do you think?

Reply:
Not sure. For this case the result would have to have it parenthesized. But I think the idea of your recipe was to only cover simple cases, so you may want to exclude the case when the argument is a "complex" expression. You currently have an exception for `J.Binary` with the `+` operator which I think is fine, but you may additionally want to check that the type of that `J.Binary` is `java.lang.String`. Then your example would not be rewritten.

Reply:
good point, adding a check that the type of that `J.Binary` is `java.lang.String` makes sense to me, will add it.